code/modules/reagents/chemistry/recipes/pyrotechnics.dm

```diff
@@ -59,23 +59,23 @@
 	if(created_volume >= 150)
 		playsound(get_turf(holder.my_atom), 'sound/effects/pray.ogg', 80, FALSE, round(created_volume/48))
 		strengthdiv = 8
+		var/deity
+		if(GLOB.deity)
+			deity = GLOB.deity
+		else
+			deity = "Christ"
 		for(var/mob/living/simple_animal/revenant/R in get_hearers_in_view(7,get_turf(holder.my_atom)))
-			var/deity
-			if(GLOB.deity)
-				deity = GLOB.deity
-			else
-				deity = "Christ"
 			to_chat(R, "<span class='userdanger'>The power of [deity] compels you!</span>")
-			R.stun(20)
+			R.stun(30)
 			R.reveal(100)
 			R.adjustHealth(50)
-		sleep(20)
-		for(var/mob/living/carbon/C in get_hearers_in_view(round(created_volume/48,1),get_turf(holder.my_atom)))
+		for(var/mob/living/carbon/C in get_hearers_in_view(7,get_turf(holder.my_atom)))
 			if(iscultist(C))
-				to_chat(C, "<span class='userdanger'>The divine explosion sears you!</span>")
+				to_chat(C, "<span class='userdanger'>The power of [deity] compels you!</span>")
 				C.Paralyze(40)
 				C.adjust_fire_stacks(5)
 				C.IgniteMob()
+		sleep(20)
```

Comment:
do not sleep, use timers.

Reply:
why are you sleeping for 20 seconds to do the base `on_reaction` to begin with?

Reply:
Hey, I didn't write this code.

Also, this is the same thing that black powder does, so I figure that it's safe to keep this in (right?).

Reply:
How would I make this use a timer, anyway?

Reply:
you need to figure out why this exists in the first place before you can figure out what timer to include